src/ddsrt/src/hopscotch.c

```diff
@@ -350,12 +349,8 @@ static void ddsrt_chh_fini (struct ddsrt_chh *rt)
 struct ddsrt_chh *ddsrt_chh_new (uint32_t init_size, ddsrt_hh_hash_fn hash, ddsrt_hh_equals_fn equals, ddsrt_hh_buckets_gc_fn gc_buckets, void *gc_buckets_arg)
 {
     struct ddsrt_chh *hh = ddsrt_malloc (sizeof (*hh));
-    if (ddsrt_chh_init (hh, init_size, hash, equals, gc_buckets, gc_buckets_arg) < 0) {
```

Comment:
I'm always a little sad when correct out-of-memory handling code gets dumped. Then I realise it's probably never going to get executed and so is just unhelpful ballast in the code ... At which point I consider that it really can happen in practice, especially on embedded machines and so that we should handle it correctly.

But then, doing it correctly for the entirety of DDS is a rather tricky thing because of the consistency that should be maintained between the different data readers (what if reader 1 gets the data and the application reads it, but it runs out of memory before it becomes visible in reader 2 in the same process?) and without wanting to incur overhead for this case that really never happens in practice ...

And so I keep switching sides. I know what I'd like: do it properly all throughout the code base, guaranteeing that it correctly handles any out-of-memory condition by aborting the operation without any observable state change and while maintaining consistency (save for some logging and statistics).

And failing that, giving up on it altogether, keeping the code free of error paths and just crashing on out-of-memory ...

This code happened to be some code that did handle it correctly, it is just that `ddsrt_malloc` aborts on out-of-memory instead of returning a null pointer (because, well, see above). I think I'd like to kick the can down the road a little longer and use `ddsrt_malloc_s` (a terribly named function if ever there was any) instead of `ddsrt_malloc`: then the error handling code is no longer dead code.

Reply:
I agree on this point. My preference is also be to maintain the out-of-memory code. It makes sense to me to operate assuming that allocation of resources may fail, while de-allocation of resources must not.

I've reverted my change and then swapped to using `ddsrt_malloc_s` in 979f8d0.

The question that remains is whether the use of `ddsrt_malloc_s` should be promoted wherever possible in the rest of the code-base.